torch/_inductor/codegen/wrapper.py

```diff
@@ -334,6 +335,59 @@ def call(args):
                 )
             V.graph.sizevars.codegen(self.wrapper_call, V.graph.graph_inputs)
 
+    def write_mkldnn_param_gen_utils(self):
+        self.header.splice(
+            """
+            class mkldnn_param_cache:
+                def __init__(self):
+                    self.p_cache = dict()
+
+                def add_param(self, param_id, param, param_type):
+                    self.p_cache[param_id] = (param, param_type)
+
+                def get_param(self, param_id):
+                    if param_id in self.p_cache:
+                        return self.p_cache[param_id][0]
```

Comment:
Can this work with multi-threaded case?

Reply:
Good catch, seems I forgot to add multi-threading support which I originally plans to add after the single-thread case works. 

As for multi-threading, the major critical area is the cache write part, I will add threading lock for the cache writing code block and use double-checking to make the impact of lock to be only on cache miss case and only generate cache item once.

And for the cache clearing, as currently we hook it to the context deleting timepoint which should be per process instead of per thread, so I'd think we don't need to add lock for the param deleting logic.